### PR TITLE
Add --strip-trailing-cr

### PR DESCRIPTION
**Pull request by Wilbeibi**

### CODE REVIEW

File: icdiff [hunk 1]
```diff
@@ -124,6 +126,13 @@ class ConsoleDiff(object):
         tolines = [expand_tabs(line) for line in tolines]
         return fromlines, tolines
 
+    def _strip_trailing_cr(self, fromlines, tolines):
+        """ Remove windows return carriage
+        """
+        fromlines = [line.replace('\r', '') for line in fromlines]
+        tolines = [line.replace('\r', '') for line in tolines]
```
Comment (jeffkaufman) [inline icdiff]: This doesn't just remove trailing `\r`, it removes `\r` everywhere

File: icdiff [hunk 2]
```diff
@@ -303,6 +312,9 @@ class ConsoleDiff(object):
         # markup
         fromlines, tolines = self._tab_newline_replace(fromlines, tolines)
 
+        if self.strip_trailing_cr:
+            fromlines, tolines = self._strip_trailing_cr(fromlines, tolines)
```
Comment (jeffkaufman) [inline icdiff]: I'd find this clearer as:

```
fromlines = self._strip_trailing_cr(fromlines)
tolines = self._strip_trailing_cr(tolines)
```

There's no reason why `_strip_trailing_cr` would want both lists of lines at once

File: icdiff [hunk 3]
```diff
@@ -487,6 +499,9 @@ def create_option_parser():
                       action="store_true",
                       help="show the whole file instead of just changed "
                       "lines and context")
+    parser.add_option("--strip-trailing-cr", default=False,
+                      action="store_true",
+                      help="ignore windows newline carriage return")
```
Comment (jeffkaufman) [inline icdiff]: This text here should match the text you added in the readme.  Either text is fine, but let's make it consistent.

File: icdiff (hunk 1 above)
Comment (Wilbeibi) [inline icdiff]: How about `line.rstrip('\r').rstrip('\r\n')` ?

**Pull request merged**